File: heads/python/src/main/python/hydra/dsl/meta/lib/literals.py
"""Phantom-typed term DSL for the hydra.lib.literals library."""

from hydra.phantoms import TTerm
from hydra.dsl.meta.phantoms import lib_primitive1 as primitive1


def bigint_to_decimal(x: TTerm) -> TTerm:
    """Convert an int to a decimal (arbitrary-precision exact decimal)."""
    return primitive1(x)


def bigint_to_int8(x: TTerm) -> TTerm:
    """Convert an int to int8."""
    return primitive1(x)


def bigint_to_int16(x: TTerm) -> TTerm:
    """Convert an int to int16."""
    return primitive1(x)


def bigint_to_int32(x: TTerm) -> TTerm:
    """Convert an int to int32."""
    return primitive1(x)


def bigint_to_int64(x: TTerm) -> TTerm:
    """Convert an int to int64."""
    return primitive1(x)


def bigint_to_uint8(x: TTerm) -> TTerm:
    """Convert an int to uint8."""
    return primitive1(x)


def bigint_to_uint16(x: TTerm) -> TTerm:
    """Convert an int to uint16."""
    return primitive1(x)


def bigint_to_uint32(x: TTerm) -> TTerm:
    """Convert an int to uint32."""
    return primitive1(x)


def bigint_to_uint64(x: TTerm) -> TTerm:
    """Convert an int to uint64."""
    return primitive1(x)


def binary_to_bytes(x: TTerm) -> TTerm:
    """Convert binary data to a list of byte values (0-255)."""
    return primitive1(x)


def binary_to_string(s: TTerm) -> TTerm:
    """Convert binary to string by decoding bytes to UTF-8."""
    return primitive1(s)


def decimal_to_bigint(x: TTerm) -> TTerm:
    """Convert a decimal to an int by truncating the fractional part."""
    return primitive1(x)


def decimal_to_float32(x: TTerm) -> TTerm:
    """Convert a decimal to a float32."""
    return primitive1(x)


def decimal_to_float64(x: TTerm) -> TTerm:
    """Convert a decimal to a float64."""
    return primitive1(x)


def float32_to_decimal(x: TTerm) -> TTerm:
    """Convert a float32 to a decimal."""
    return primitive1(x)


def float32_to_float64(x: TTerm) -> TTerm:
    """Convert a float32 to a float64 (lossless widening)."""
    return primitive1(x)


def float64_to_decimal(x: TTerm) -> TTerm:
    """Convert a float64 to a decimal."""
    return primitive1(x)


def float64_to_float32(x: TTerm) -> TTerm:
    """Convert a float64 to a float32 (lossy narrowing)."""
    return primitive1(x)


def int8_to_bigint(x: TTerm) -> TTerm:
    """Convert int8 to int."""
    return primitive1(x)


def int16_to_bigint(x: TTerm) -> TTerm:
    """Convert int16 to int."""
    return primitive1(x)


def int32_to_bigint(x: TTerm) -> TTerm:
    """Convert int32 to int."""
    return primitive1(x)


def int64_to_bigint(x: TTerm) -> TTerm:
    """Convert int64 to int."""
    return primitive1(x)


def read_boolean(s: TTerm) -> TTerm:
    """Parse a string to a boolean."""
    return primitive1(s)


def read_decimal(s: TTerm) -> TTerm:
    """Parse a string to a decimal (arbitrary-precision exact decimal)."""
    return primitive1(s)


def read_float32(s: TTerm) -> TTerm:
    """Parse a string to a float."""
    return primitive1(s)


def read_float64(s: TTerm) -> TTerm:
    """Parse a string to a float."""
    return primitive1(s)


def read_int32(s: TTerm) -> TTerm:
    """Parse a string to an int."""
    return primitive1(s)


def read_int64(s: TTerm) -> TTerm:
    """Parse a string to an int."""
    return primitive1(s)


def read_string(s: TTerm) -> TTerm:
    """Parse a string literal."""
    return primitive1(s)


def show_bigint(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_boolean(b: TTerm) -> TTerm:
    """Convert a boolean to string."""
    return primitive1(b)


def show_decimal(x: TTerm) -> TTerm:
    """Convert a decimal (arbitrary-precision exact decimal) to its string representation."""
    return primitive1(x)


def show_float32(x: TTerm) -> TTerm:
    """Convert a float to string."""
    return primitive1(x)


def show_float64(x: TTerm) -> TTerm:
    """Convert a float to string."""
    return primitive1(x)


def show_int8(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_int16(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_int32(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_int64(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_uint8(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_uint16(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_uint32(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_uint64(x: TTerm) -> TTerm:
    """Convert an int to string."""
    return primitive1(x)


def show_string(s: TTerm) -> TTerm:
    """Convert a string to a quoted string representation."""
    return primitive1(s)


def string_to_binary(s: TTerm) -> TTerm:
    """Convert string to binary by encoding string to UTF-8 bytes."""
    return primitive1(s)


def uint8_to_bigint(x: TTerm) -> TTerm:
    """Convert uint8 to int."""
    return primitive1(x)


def uint16_to_bigint(x: TTerm) -> TTerm:
    """Convert uint16 to int."""
    return primitive1(x)


def uint32_to_bigint(x: TTerm) -> TTerm:
    """Convert uint32 to int."""
    return primitive1(x)


def uint64_to_bigint(x: TTerm) -> TTerm:
    """Convert uint64 to int."""
    return primitive1(x)

File: packages/hydra-java/src/main/java/hydra/dsl/meta/lib/Literals.java
package hydra.dsl.meta.lib;
import hydra.dsl.meta.Phantoms;
import hydra.typed.TypedTerm;

/** Phantom-typed term DSL for the {@code hydra.lib.literals} library. */
public final class Literals {
    private Literals() {}

    private static <A> TypedTerm<A> prim(String name) {
        return Phantoms.var("hydra.lib.literals." + name);
    }

    public static <A> TypedTerm<A> bigintToDecimal(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToDecimal"), x); }
    public static <A> TypedTerm<A> bigintToInt8(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToInt8"), x); }
    public static <A> TypedTerm<A> bigintToInt16(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToInt16"), x); }
    public static <A> TypedTerm<A> bigintToInt32(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToInt32"), x); }
    public static <A> TypedTerm<A> bigintToInt64(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToInt64"), x); }
    public static <A> TypedTerm<A> bigintToUint8(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToUint8"), x); }
    public static <A> TypedTerm<A> bigintToUint16(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToUint16"), x); }
    public static <A> TypedTerm<A> bigintToUint32(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToUint32"), x); }
    public static <A> TypedTerm<A> bigintToUint64(TypedTerm<?> x) { return Phantoms.apply(prim("bigintToUint64"), x); }
    public static <A> TypedTerm<A> binaryToBytes(TypedTerm<?> x) { return Phantoms.apply(prim("binaryToBytes"), x); }
    public static <A> TypedTerm<A> binaryToString(TypedTerm<?> s) { return Phantoms.apply(prim("binaryToString"), s); }
    public static <A> TypedTerm<A> decimalToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("decimalToBigint"), x); }
    public static <A> TypedTerm<A> decimalToFloat32(TypedTerm<?> x) { return Phantoms.apply(prim("decimalToFloat32"), x); }
    public static <A> TypedTerm<A> decimalToFloat64(TypedTerm<?> x) { return Phantoms.apply(prim("decimalToFloat64"), x); }
    public static <A> TypedTerm<A> float32ToDecimal(TypedTerm<?> x) { return Phantoms.apply(prim("float32ToDecimal"), x); }
    public static <A> TypedTerm<A> float32ToFloat64(TypedTerm<?> x) { return Phantoms.apply(prim("float32ToFloat64"), x); }
    public static <A> TypedTerm<A> float64ToDecimal(TypedTerm<?> x) { return Phantoms.apply(prim("float64ToDecimal"), x); }
    public static <A> TypedTerm<A> float64ToFloat32(TypedTerm<?> x) { return Phantoms.apply(prim("float64ToFloat32"), x); }
    public static <A> TypedTerm<A> int8ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("int8ToBigint"), x); }
    public static <A> TypedTerm<A> int16ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("int16ToBigint"), x); }
    public static <A> TypedTerm<A> int32ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("int32ToBigint"), x); }
    public static <A> TypedTerm<A> int64ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("int64ToBigint"), x); }
    public static <A> TypedTerm<A> readBoolean(TypedTerm<?> s) { return Phantoms.apply(prim("readBoolean"), s); }
    public static <A> TypedTerm<A> readDecimal(TypedTerm<?> s) { return Phantoms.apply(prim("readDecimal"), s); }
    public static <A> TypedTerm<A> readFloat32(TypedTerm<?> s) { return Phantoms.apply(prim("readFloat32"), s); }
    public static <A> TypedTerm<A> readFloat64(TypedTerm<?> s) { return Phantoms.apply(prim("readFloat64"), s); }
    public static <A> TypedTerm<A> readInt32(TypedTerm<?> s) { return Phantoms.apply(prim("readInt32"), s); }
    public static <A> TypedTerm<A> readInt64(TypedTerm<?> s) { return Phantoms.apply(prim("readInt64"), s); }
    public static <A> TypedTerm<A> readString(TypedTerm<?> s) { return Phantoms.apply(prim("readString"), s); }
    public static <A> TypedTerm<A> showBigint(TypedTerm<?> x) { return Phantoms.apply(prim("showBigint"), x); }
    public static <A> TypedTerm<A> showBoolean(TypedTerm<?> b) { return Phantoms.apply(prim("showBoolean"), b); }
    public static <A> TypedTerm<A> showDecimal(TypedTerm<?> x) { return Phantoms.apply(prim("showDecimal"), x); }
    public static <A> TypedTerm<A> showFloat32(TypedTerm<?> x) { return Phantoms.apply(prim("showFloat32"), x); }
    public static <A> TypedTerm<A> showFloat64(TypedTerm<?> x) { return Phantoms.apply(prim("showFloat64"), x); }
    public static <A> TypedTerm<A> showInt8(TypedTerm<?> x) { return Phantoms.apply(prim("showInt8"), x); }
    public static <A> TypedTerm<A> showInt16(TypedTerm<?> x) { return Phantoms.apply(prim("showInt16"), x); }
    public static <A> TypedTerm<A> showInt32(TypedTerm<?> x) { return Phantoms.apply(prim("showInt32"), x); }
    public static <A> TypedTerm<A> showInt64(TypedTerm<?> x) { return Phantoms.apply(prim("showInt64"), x); }
    public static <A> TypedTerm<A> showUint8(TypedTerm<?> x) { return Phantoms.apply(prim("showUint8"), x); }
    public static <A> TypedTerm<A> showUint16(TypedTerm<?> x) { return Phantoms.apply(prim("showUint16"), x); }
    public static <A> TypedTerm<A> showUint32(TypedTerm<?> x) { return Phantoms.apply(prim("showUint32"), x); }
    public static <A> TypedTerm<A> showUint64(TypedTerm<?> x) { return Phantoms.apply(prim("showUint64"), x); }
    public static <A> TypedTerm<A> showString(TypedTerm<?> s) { return Phantoms.apply(prim("showString"), s); }
    public static <A> TypedTerm<A> stringToBinary(TypedTerm<?> s) { return Phantoms.apply(prim("stringToBinary"), s); }
    public static <A> TypedTerm<A> uint8ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("uint8ToBigint"), x); }
    public static <A> TypedTerm<A> uint16ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("uint16ToBigint"), x); }
    public static <A> TypedTerm<A> uint32ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("uint32ToBigint"), x); }
    public static <A> TypedTerm<A> uint64ToBigint(TypedTerm<?> x) { return Phantoms.apply(prim("uint64ToBigint"), x); }
}
